src/components/Graph/Graph.tsx

```diff
@@ -90,30 +89,24 @@ const Graph: FC<GraphProps> = memo(({ interactive = true, expandCurrentNode = tr
       if (edge.label) edge.label.x += toAdd;
     });
   }
-  const layoutKeys = Object.keys(layout).toString();
-  const initialExpandedState = useMemo(() => {
-    return layoutKeys.split(',').reduce((acc: { [key: string]: boolean }, curr: string) => {
-      acc[curr] = false;
-      return acc;
-    }, {});
-  }, [layoutKeys]);
 
-  const [expanded, _setExpanded] = useState({
-    expandedMap: initialExpandedState,
-    lastSelectedNode: ''
-  });
+  const [expanded, setExpanded] = useState<ExpandedState>(() =>
+    Object.keys(layout).reduce(
+      (acc, curr: string) => {
+        acc[curr] = false;
+        return acc;
+      },
+      { currentNode: null, lastSelectedNode: null } as ExpandedState
+    )
+  );
 
-  const setExpanded = useCallback((key: string): void => {
-    _setExpanded(prevState => {
-      let newExpandedMap = prevState.expandedMap;
-      if (!prevState.expandedMap[key] || prevState.lastSelectedNode === key) {
-        newExpandedMap = {
-          ...prevState.expandedMap,
-          [key]: !prevState.expandedMap[key]
-        };
-      }
-      return { ...prevState, expandedMap: newExpandedMap, lastSelectedNode: key };
-    });
+  const toggleExpanded = useCallback((key: string) => {
+    setExpanded(prevState => ({
+      ...prevState,
+      [key]: !prevState[key] || prevState.lastSelectedNode === key,
```

Comment:
It's not immediately obvious to me why the behavior is different, but I think it's this line -- switching to an expanded node "A" from a node that was just collapsed on the previous click "B" now collapses node A instead of just switching to it.
Steps:
1. Click on a node A to open & select it
2. Click on a node B to open & select it, A is still open
3. Click on B again to close it, A is still open
4. Click on A again.
Before this would leave A open and select it, now it collapses A.
We do have a task in the backlog (257) for reviewing the interaction in more detail so it's not critical, but if we can change one line and bring it back to how it was before that would be great

Reply:
Good catch! I just pushed up the fix.

Reply:
I'm still observing some weird behavior with the nodes. With two nodes open it seems to work fine but with more than 2 nodes open sometimes switching to a node will select it and close it. Haven't taken a look at the code to see why but wanted to mention it. 

(Github won't let me upload the video here but I added it as a comment to this PR in Slack: https://mitrecorp.slack.com/archives/CLH34KL6A/p1589232347007300?thread_ts=1589058599.028800&cid=CLH34KL6A)

Reply:
Pushed up another fix. Hopefully this now has the same behavior as master.